src/db/collections/analytics/ledger_size.rs

```diff
@@ -64,30 +60,63 @@ impl OutputCollection {
         &self,
         ledger_index: MilestoneIndex,
     ) -> Result<LedgerSizeAnalyticsResult, Error> {
-        Ok(self
-        .aggregate(
-            vec![
-                doc! { "$match": {
-                    "metadata.booked.milestone_index": { "$lte": ledger_index },
-                    "metadata.spent_metadata.spent.milestone_index": { "$not": { "$lte": ledger_index } }
-                } },
-                doc! { "$group" : {
-                    "_id": null,
-                    "total_key_bytes": { "$sum": { "$toDecimal": "$details.rent_structure.num_key_bytes" } },
-                    "total_data_bytes": { "$sum": { "$toDecimal": "$details.rent_structure.num_data_bytes" } },
-                    "total_storage_deposit_value": { "$sum": { "$toDecimal": "$output.storage_deposit_return_unlock_condition.amount" } },
-                } },
-                doc! { "$project": {
-                    "total_storage_deposit_value": { "$toString": "$total_storage_deposit_value" },
-                    "total_key_bytes": { "$toString": "$total_key_bytes" },
-                    "total_data_bytes": { "$toString": "$total_data_bytes" },
-                } },
-            ],
-            None,
-        )
-        .await?
-        .try_next()
-        .await?
-        .unwrap_or_default())
+        #[derive(Deserialize)]
+        struct Res {
+            total_storage_deposit_value: String,
+            total_key_bytes: String,
+            total_data_bytes: String,
+            rent_structure: RentStructure,
+        }
+
+        let res = self
+            .aggregate::<Res>(
+                vec![
+                    doc! { "$match": {
+                        "metadata.booked.milestone_index": { "$lte": ledger_index },
+                        "metadata.spent_metadata.spent.milestone_index": { "$not": { "$lte": ledger_index } }
+                    } },
+                    doc! { "$group" : {
+                        "_id": null,
+                        "total_key_bytes": { "$sum": { "$toDecimal": "$details.rent_structure.num_key_bytes" } },
+                        "total_data_bytes": { "$sum": { "$toDecimal": "$details.rent_structure.num_data_bytes" } },
+                        "total_storage_deposit_value": { "$sum": { "$toDecimal": "$output.storage_deposit_return_unlock_condition.amount" } },
+                    } },
+                    doc! { "$lookup": {
+                        "from": ProtocolUpdateCollection::NAME,
+                        "pipeline": [
+                            { "$match": { "_id": { "$lte": ledger_index } } },
```

Comment:
This would return the first it finds that is `< ledger_index`, but not necessarily the closest, right? Feels like this could become a problem.

Reply:
Hm. Good catch.